hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnection.java

```diff
@@ -246,14 +233,13 @@ private void logClose() {
     }
 
     protected void innerClose() throws IOException {
-        // todo: closing needs to be improved because we can end up with some channels not closed.
-        channel.close();
-
-        if (tpcChannels != null) {
-            for (Channel tpcChannel : tpcChannels) {
-                tpcChannel.close();
+        if (altoChannels != null) {
+            for (Channel altoChannel : altoChannels) {
+                closeResource(altoChannel);
```

Comment:
closeResource swallows any close errors, will the channels stay open if that is the case?

Reply:
Well, if the channel throws while we try to close it, there is not much we can do other than swallow it, and continue closing the other channels. If you take a look at the current situation, it is the same there as well. In the master, if the innerClose method throws during the connection#close call, we just log the exception and continue. 

But, let me update this to use the logger of this class, and do a log with the warning level, just like the connection.